### PR TITLE
Torsorial type families

### PR DESCRIPTION
**Pull request by EgbertRijke**
This PR adds:

- torsorial type families
- pointed torsorial type families
- sorial type families
- join powers of types

following a project that I'm doing with Buchholtz, Christensen, Myers, and Taxeras Flaten.

### CODE REVIEW

**Comment by fredrik-bakke** [comment]
Do you want this PR reviewed?

**Comment by EgbertRijke** [comment]
Sure! This stuff comes from a paper that I'm working on with Dan Christensen, David Jaz Myers, Ulrik Buchholtz, and Jarl Taxeras Flaten.

But wait a bit, because I just thought of some nice additions:)

**Comment by EgbertRijke** [comment]
I'm formalizing the old fact that `fib Id B` is equivalent to `is-contr (Sigma A B)`, which fits perfectly in the story of torsorial type families.

**Comment by EgbertRijke** [comment]
The fact I wanted to add is added, and the links are checked. This is ready for review:)

File: src/foundation/sorial-type-families.lagda.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+### Sorial type families
+
+```agda
+module _
+  {l1 l2 : Level} (B : Pointed-Type l1) (E : type-Pointed-Type B → UU l2)
+  where
+
+  is-sorial-family-of-types : UU (l1 ⊔ l2)
+  is-sorial-family-of-types =
```
Comment (fredrik-bakke) [inline src/foundation/sorial-type-families.lagda.md:42]: In some places, names like `is-sorial-family` are used instead. We should perhaps be consistent on this, although I think your naming is fine.

File: src/foundation/universal-property-identity-types.lagda.md [hunk 2]
```diff
@@ -58,3 +72,132 @@ equiv-ev-refl' :
...
+
+#### Axiom L implies that `Id : A → (A → 𝒰)` is an embedding
+
+The proof that axiom L implies that `Id : A → (A → 𝒰)` is an embedding proceeds
+via the
+[fundamental theorem of identity types](foundation.fundamental-theorem-of-identity-types.md)
+by showing that the [fiber](foundation.fibers-of-maps.md) of `Id` at `Id a` is
+[contractible](foundation.contractible-types.md) for each `a : A`. To see this,
+we first note that this fiber has an element `(a , refl)`. Therefore it suffices
+to show that thsi fiber is a proposition. We do this by constructing an
```
Comment (fredrik-bakke) [inline src/foundation/universal-property-identity-types.lagda.md:91]: this*

File: src/synthetic-homotopy-theory/join-powers-of-types.lagda.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+```agda
+join-power : {l1 : Level} → ℕ → UU l1 → UU l1
+join-power n A = iterate n (join A) (raise-empty _)
+```
+
+### Join powers of type families
+
+```agda
+join-power-family-of-types :
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/join-powers-of-types.lagda.md:39]: Same comment as before

**Review by fredrik-bakke — APPROVED** [review]
An almost flawless PR. Very well done! 😁

**Comment by fredrik-bakke** [comment]
This theory of torsorial types is intriguing, especially with the connection to Axiom L. Looking forward to your next PRs/the upcoming paper!

File: src/foundation/sorial-type-families.lagda.md (hunk 1 above)
Comment (EgbertRijke) [reply src/foundation/sorial-type-families.lagda.md:42]: I think I only did that in file names. Would you like me to change them?

Comment (fredrik-bakke) [reply src/foundation/sorial-type-families.lagda.md:42]: Searching for `-family :` tells me that there are at least 19 definitions using this convention, as opposed to zero for `-family-of-types :`. You have used `family-of-ideals` before though.

Comment (EgbertRijke) [reply src/foundation/sorial-type-families.lagda.md:42]: What is your command to search for these?

Comment (fredrik-bakke) [reply src/foundation/sorial-type-families.lagda.md:42]: `CMD+Shift+F` (in VSCode on Mac)

Comment (EgbertRijke) [reply src/foundation/sorial-type-families.lagda.md:42]: lol, I gotta fire up the VScode then:)

Haha, I was hoping for a command for the terminal

Comment (EgbertRijke) [reply src/foundation/sorial-type-families.lagda.md:42]: I see that we're very inconsistent with this: we have `family`, `family-of-types`, and `type-family` in names. Can I change this in another PR?

Comment (fredrik-bakke) [reply src/foundation/sorial-type-families.lagda.md:42]: Sure! I'm fine with that :)

**Pull request merged**